features/FxA-71-AMO-migration-support/README.md

```diff
@@ -0,0 +1,189 @@
+
+Migration support for [AMO](https://addons.mozilla.org/)
+========================================================
+
+https://mozilla.aha.io/features/FXA-71
+
+## Stories
+
+As a developer of addons.mozilla.org,
+I want to migrate user logins from site-specific passwords
+to Firefox Accounts,
+so that users have fewer credentials to remember
+across Mozilla's online properties.
+
+As an existing user of addons.mozilla.org,
+I want to clearly understand how and why
+I should migrate to a Firefox Account,
+so I can continue using the site
+with minimal interruption.
+
+
+## Measuring Success
+
+We will measure success of this feature
+by tracking whether users are successful
+at completing the migration.
+The key metric will be
+the number of users who successfully sign in or sign up,
+as a percentage of the number of users
+who enter the FxA login flow from AMO.
+
+
+## UX Mocks
+
+The overall UX for the migration flow
+was originally specified in https://github.com/mozilla/addons/pull/9.
+This feature captures the FxA-specific components
+so that we can properly manage and track their implementation.
+
+Three changes to our existing flows are required.
+
+### Migration indicator
+
+For users landing on the signup page
+from the AMO migration flow,
+we should show a highlight to inform them
+that they can use an existing account
+if they have one:
+
+![](migration-indicator.png)
+
+### Permission prompt
+
+Before redirecting back to AMO,
```

Comment:
This section is confusing - don't users select the permissions just after the signup screen? The wording here implies post-verification.

If post-verification, let's work through the possible scenarios:

1. User signs up in tab 1 of browser A, verifies in same tab (tab 1) of browser A.
1. User signs up in tab 1 of browser A, verifies in new tab (tab 2) of browser A with tab 1 still open.
1. User signs up in tab 1 of browser A, verifies in new tab (tab 2) of browser A with tab 1 closed.
1. User signs up in tab 1 of browser A, verifies in browser B with tab 1 from browser A still open.
1. User signs up in tab 1 of browser A, verifies in browser B with tab 1 from browser A closed.

In each of these scenarios, which tab redirects back to AMO? A previous requirement from Marketplace was that only the original tab redirects. Does AMO have this same restriction? 

If the permission screen is displayed post-verification, where is that shown? Tab 1, or tab 2? Tab 1 is problematic because we have no way to force the user back to the original tab.